homeassistant/helpers/entity_registry.py

```diff
@@ -201,6 +204,14 @@ def async_generate_entity_id(
         Conflicts checked against registered and currently existing entities.
         """
         preferred_string = f"{domain}.{slugify(suggested_object_id)}"
+
+        for param, param_name, max_length in (
```

Comment:
The check at the end is probably all that is needed.  It should be very rare that this hits